controller_manager:
  ros__parameters:
    update_rate: 1000 #Hz

    # Publish all joint states -----------------------------------
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # Velocity Controllers ---------------------------------------
    left_rear_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController

    right_rear_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController

    left_front_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController

    right_front_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController

    # Position Controllers ---------------------------------------
    left_steering_hinge_position_controller:
      type: effort_controllers/JointPositionController
    right_steering_hinge_position_controller:
      type: effort_controllers/JointPositionController


left_rear_wheel_velocity_controller:
  ros__parameters:
    joints: 
      -left_rear_wheel_joint
    # pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
    state_interfaces: velocity
    command_interfaces: effort
right_rear_wheel_velocity_controller:
  ros__parameters:
    joint: 
      -right_rear_wheel_joint
    # pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
    state_interfaces: velocity
    command_interfaces: effort
left_front_wheel_velocity_controller:
  ros__parameters:
    joints: 
      -left_front_wheel_joint
    # pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}
    state_interfaces: velocity
    command_interfaces: effort
right_front_wheel_velocity_controller:
  ros__parameters:
    joints: 
      -right_front_wheel_joint
    # pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}
    state_interfaces: velocity
    command_interfaces: effort


# Position Controllers ---------------------------------------
left_steering_hinge_position_controller:
  ros__parameters:
    joints: 
      -left_steering_hinge_joint
    # pid: {p: 1.0, i: 0.0, d: 0.5}
    state_interfaces: position
    command_interfaces: effort
right_steering_hinge_position_controller:
  ros__parameters:
    joints: 
      -right_steering_hinge_joint
    # pid: {p: 1.0, i: 0.0, d: 0.5}
    state_interfaces: position
    command_interfaces: effort

